### PR TITLE
Add event that's triggered when slider is released

### PR DESCRIPTION
**Pull request by freddyaboulton**
# Description

Took me forever to figure out how to properly test the front-end/back-end flow via end-to-end test 😅 

I added a test that spins up a backend and then programmatically modifies the input slider and gets the response from the backend. I think this is a useful pattern that we should leverage to increase automated test coverage.

Summary of change:

- Added a `release` EventListener class that lets users register functions for the slider's new "release" event. I originally went with my own suggestion to use the `blur` event but I don't think it's intuitive for users to have to click outside the slider to launch the event.
- Necessary front end modifications + end-to-end test.

Should the `release` event also be modified if the slider's typeable value field is changed? 

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added a short summary of my change to the CHANGELOG.md
- [x] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md, unless the pull request is labeled with the "no-changelog-update" label.

Please add a brief summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

**Comment by pngwn** [comment]
> I added a test that spins up a backend and then programmatically modifies the input slider and gets the response from the backend. I think this is a useful pattern that we should leverage to increase automated test coverage.

The reason we don't do this is because it scales poorly and will make the tests too slow overall. We only need one or two 'complete' end-to-end tests and they should cut across most of the API in a single run. For everything else I'd like to us to carry on mocking the APIs or just writing unit tests.

We do need to write those end-to-end tests though, feel free to take those on if you fancy it.

**Comment by freddyaboulton** [comment]
Yea agreed we shouldn't spin up a separate backend for to test a single feature. On board with your suggestion of spinning up one server with one demo (or demos) running that should enable us to test most of the api in an end-to-end fashion. Happy to take that on by extending the `slider-release` test to cover more of the api. I think it will help make gradio more stable by catching unintended regressions prior to release.

**Comment by pngwn** [comment]
Yeah, we definitely need to add some real end-to-end tests. Would be awesome.

File: gradio/events.py [hunk 1]
```diff
@@ -721,3 +721,60 @@ def upload(
             every=every,
         )
         set_cancel_events(self, "upload", cancels)
+
```
Comment (pngwn) [inline gradio/events.py:724]: I still think we should rethink this, feels like a lot of code when it is borderline identical to all of the other events,

File: ui/packages/app/test/slider_release.spec.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	const endPoint = {
+		x: sliderBoundingBox.x + sliderBoundingBox.width * targetPercentage,
+		y: thumbBoundingBox.y + thumbBoundingBox.height / 2
+	};
+
+	await page.mouse.move(startPoint.x, startPoint.y);
+	await page.mouse.down();
+	await page.mouse.move(endPoint.x, endPoint.y);
+	await page.mouse.up();
+}
```
Comment (pngwn) [inline ui/packages/app/test/slider_release.spec.ts:33]: magic

File: ui/packages/app/test/slider_release.spec.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	const responsePromise = page.waitForResponse(
+		"http://127.0.0.1:7888/run/predict/"
+	);
+	await changeSlider(page, slider, slider, 0.7);
+	const response = await responsePromise;
+	const responseData = await response.json();
+
+	expect(responseData.data[0]).toBeGreaterThan(69.5);
+	expect(responseData.data[0]).toBeLessThan(71.0);
+	//const eventTracker = page.getByLabel("Number of events fired");
```
Comment (pngwn) [inline ui/packages/app/test/slider_release.spec.ts:49]: Remove this.

**Review by pngwn — APPROVED** [review]
lgtm! just a leftover comment and everything looks fine.

File: gradio/events.py (hunk 1 above)
Comment (freddyaboulton) [reply gradio/events.py:724]: Agreed 100%

**Pull request merged**